source/designers-guide/smarty/index.md

```diff
@@ -199,6 +199,35 @@ In Shopware 5 we added the ability to register your own custom Smarty plugins in
 **Example:** *Path to custom Smarty plugins*
 
 ```
-/_private/smarty/function.markdown.php
-/_private/smarty/modifier.picture.php
+themes/Frontend/ThemeName/_private/smarty/function.markdown.php
+themes/Frontend/ThemeName/_private/smarty/modifier.picture.php
+```
+**Notice:** Smarty plugins are not working inside mail templates
+
+#### Register custom smarty plugins in shopware plugin ####
+
+To specify your custom plugins directory inside your shopware plugin you can use the `CompilerPass` to add your own directory:
+
+```php
+<?php
+namespace PluginName\Components\CompilerPass;
+
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+/**
+ * Class AddTemplatePluginDirCompilerPass
+ */
+class AddTemplatePluginDirCompilerPass implements CompilerPassInterface
```

Comment:
is this necessary? afaik, there is also a convention how the directory has to look like 

@shyim to the rescue?